pyomo/contrib/fme/plugins/fourier_motzkin_elimination.py

```diff
@@ -0,0 +1,329 @@
+                # assumption below.
+                # NOTE: neither of the scalar multiplications below flip the
+                # constraint. So we are sure to have only geq constraints
+                # forever, which is exactly what we want.
+                if leaving_var_coef < 0:
+                    leq_list.append(
+                        self._nonneg_scalar_multiply_linear_constraint(
+                            cons, -1.0/leaving_var_coef))
+                else:
+                    geq_list.append(
+                        self._nonneg_scalar_multiply_linear_constraint(
+                            cons, 1.0/leaving_var_coef))
+
+            for leq in leq_list:
+                for geq in geq_list:
+                    constraints.append(self._add_linear_constraints(leq, geq))
+
+            # add back in the constraints that didn't have the variable we were
+            # projecting out
+            constraints.extend(waiting_list)
+
+        return constraints
+
+    def _nonneg_scalar_multiply_linear_constraint(self, cons, scalar):
+        """Multiplies all coefficients and the RHS of a >= constraint by scalar.
+        There is no logic for flipping the equality, so this is just the 
+        special case with a nonnegative scalar, which is all we need.
+        """
+        cons['body'].linear_coefs = [scalar*coef for coef in
+                                     cons['body'].linear_coefs]
+        # and update the map... (It isn't lovely that I am storing this in two
+        # places...)
+        for var, coef in cons['map'].items():
+            cons['map'][var] = coef*scalar
+
+        # assume scalar >= 0 and constraint only has lower bound
+        if cons['lower'] is not None:
+            cons['lower'] *= scalar
+        
+        return cons
+
+    def _add_linear_constraints(self, cons1, cons2):
+        """Adds two >= constraints"""
+        ans = {'lower': None, 'upper': None, 'body': None, 'map': ComponentMap()}
+
+        # This is not beautiful, but it needs to be both deterministic and
+        # account for the fact that Vars aren't hashable.
+        seen = ComponentSet()
+        all_vars = []
+        for v in cons1['body'].linear_vars:
+            all_vars.append(v)
+            seen.add(v)
+        for v in cons2['body'].linear_vars:
+            if v not in seen:
+                all_vars.append(v)
+        
+        expr = 0
+        for var in all_vars:
+            cons1_coef = cons1['map'].get(var)
+            cons2_coef = cons2['map'].get(var)
+            if cons2_coef is not None and cons1_coef is not None:
+                ans['map'][var] = new_coef = cons1_coef + cons2_coef
+            elif cons1_coef is not None:
+                ans['map'][var] = new_coef = cons1_coef
+            elif cons2_coef is not None:
+                ans['map'][var] = new_coef = cons2_coef
+            expr += new_coef*var
+        ans['body'] = generate_standard_repn(expr)
```

Comment:
Can this be simplified to
```python
ans['body'] = generate_standard_repn(sum(
    (cons1['map'].get(var,0) + cons2['map'].get(var,0))*var for var in all_vars))
```

Reply:
No - `+=` shouldn't be a problem.  I just missed that you were updating the `map` as well in my rush to simplify things.